src/main/java/com/synopsys/integration/alert/web/security/AuthenticationHandler.java

```diff
@@ -22,41 +22,380 @@
+        this.sslValidator = sslValidator;
+        this.configurationAccessor = configurationAccessor;
+        this.csrfTokenRepository = csrfTokenRepository;
+    }
+
+    @Bean
+    public static SAMLBootstrap sAMLBootstrap() {
+        return new SAMLBootstrap();
+    }
+
+    @Override
+    protected void configure(final AuthenticationManagerBuilder auth) throws Exception {
+        auth.authenticationProvider(samlAuthenticationProvider());
     }
 
     @Override
     protected void configure(final HttpSecurity http) throws Exception {
-        ignorePaths(H2_CONSOLE_PATH, RESET_PASSWORD_PATH, RESET_PASSWORD_WITH_USERNAME_PATH);
-        httpPathManager.completeHttpSecurity(http);
-        // The profile above ensures that this will not be used if SSL is enabled.
-        http.headers().frameOptions().disable();
+        http.exceptionHandling().authenticationEntryPoint(samlEntryPoint());
+        if (sslValidator.isSSLEnabled()) {
+            configureWithSSL(http);
+        } else {
+            configureInsecure(http);
+        }
+        final RequestMatcher[] ignoringRequestMatchers = createCsrfIgnoreMatchers();
+        final RequestMatcher[] authorizedRequestMatchers = createAllowedPathMatchers();
+        http.csrf().csrfTokenRepository(csrfTokenRepository).ignoringRequestMatchers(ignoringRequestMatchers)
+            .and().headers().frameOptions().disable()
+            .and().addFilterBefore(metadataGeneratorFilter(), ChannelProcessingFilter.class)
+            .addFilterAfter(samlFilter(), BasicAuthenticationFilter.class)
+            .authorizeRequests().requestMatchers(authorizedRequestMatchers).permitAll()
+            .and().authorizeRequests().anyRequest().hasRole(UserRole.ALERT_ADMIN.name())
+            .and().logout().logoutSuccessUrl("/");
+    }
+
+    private void configureInsecure(final HttpSecurity http) throws Exception {
+        ignorePaths(HttpPathManager.PATH_H2_CONSOLE);
+    }
+
+    private void configureWithSSL(final HttpSecurity http) throws Exception {
+        http.requiresChannel().anyRequest().requiresSecure();
     }
 
     private void ignorePaths(final String... paths) {
         for (final String path : paths) {
             httpPathManager.addAllowedPath(path);
             httpPathManager.addCsrfIgnoredPath(path);
+            httpPathManager.addSamlAllowedPath(path);
+            httpPathManager.addSamlCsrfIgnoredPath(path);
         }
     }
 
+    private RequestMatcher[] createCsrfIgnoreMatchers() {
+        final RequestMatcher[] matchers = {
+            new SamlAntMatcher(samlContext(), httpPathManager.getSamlCsrfIgnoredPaths(), httpPathManager.getCsrfIgnoredPaths())
+        };
+        return matchers;
+    }
+
+    private RequestMatcher[] createAllowedPathMatchers() {
+        final RequestMatcher[] matchers = {
+            new SamlAntMatcher(samlContext(), httpPathManager.getSamlAllowedPaths(), httpPathManager.getAllowedPaths())
+        };
+        return matchers;
+    }
```

Comment:
This class is getting pretty large. Would we be able to separate out the Beans from here into a different configuration class?

Reply:
AuthenticationConfiguration

Reply:
Unfortunately you can't because to create one of the SAML beans you need to call a method that only exists in this class.  I tried to separate it out once already to do exactly what you suggested.  Line 221 the authenticationManager() method is the issue.